src/natalie.cpp

```diff
@@ -511,6 +511,18 @@ void handle_top_level_exception(Env *env, ExceptionObject *exception, bool run_e
         } else {
             clean_up_and_exit(1);
         }
+    } else if (exception->is_a(env, find_top_level_const(env, "SignalException"_s)->as_class())) {
+        Value signo = exception->ivar_get(env, "@signo"_s);
+        if (signo->type() == Object::Type::Integer) {
+            nat_int_t val = signo->as_integer()->to_nat_int_t();
+            if (val >= 0 && val <= 255) {
+                clean_up_and_exit(val);
+            } else {
+                clean_up_and_exit(1);
+            }
+        } else {
+            clean_up_and_exit(1);
+        }
```

Comment:
I think this part is incorrect. When I run CRuby with gdb, the program actually terminates with the signal from the SignalException, which is what I would expect.

![Screenshot 2023-12-30 7 24 00 AM](https://github.com/natalie-lang/natalie/assets/669/369bb7ad-d0e2-404d-afa5-a836225e2228)

Reply:
Maybe we need to `::kill(::getpid(), signal)` or some such?

Reply:
After thinking about this some more, I think what you have here is fine for now. I have in mind to work on signal handling soon anyway, so I can work on it then...

Reply:
I just pushed a fix to get the correct exit value. It should be good enough to go now.
This change is very much a chicken and egg with Process.kill and Signal.trap, I guess we have to start somewhere.